### ISSUE TITLE
False negative when passing an arbitrary value to a function that takes an argument of type `T`

### ISSUE THREAD

**Issue by LeeeeT** [microsoft/pyright#7718]
Pyright should produce an error for the following code, as `object` cannot be assigned to `T@f`.

```python
from collections.abc import Callable

def f[T](v: object, c: Callable[[], Callable[[T], object]]) -> None:
    c()(v)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.364.
